### PR TITLE
Support for OpenID Connect authentication added

### PR DESCRIPTION
**Pull request by Nerade**

### CODE REVIEW

**Comment by Nerade** [comment]
If I understood correctly there is a difference between OpenID and OpenID Connect. Am I wrong on this one?

**Comment by yaamai** [comment]
I want to use CodiMD with custom OpenID Connect provider (like keycloak).
I think, this PR is implemented wanted feature.

Currently implemented "OpenID" feature can configured with custom provider?

**Comment by Amejonah1200** [comment]
@SISheogorath OpenID and OpenID Connect are 2 different things.
Also, based upon reading the code, I cannot find on how to setup a custom OIDC provider to auth.
OAuth2 exists, yes, but it doesn't work with https://docs.gitlab.com/api/oauth2/ where all necessary env vars are provided.

If OpenID is used: "Failed to discover OP Endpoint URL"
If OAuth2 is used: "Failed to fetch user profile"

File: app.js [hunk 1]
```diff
@@ -195,6 +195,7 @@ app.locals.authProviders = {
   saml: config.isSAMLEnable,
   oauth2: config.isOAuth2Enable,
   oauth2ProviderName: config.oauth2.providerName,
+  oidc: config.isOidcnable,
```
Comment (Amejonah1200) [inline app.js:198]: Typo.

**Pull request closed without merging**